src/main/java/net/javadiscord/javabot/systems/qotw/submissions/SubmissionManager.java

```diff
@@ -122,37 +123,78 @@ private boolean canCreateSubmissions(Member member) {
 		if (member == null) return false;
 		if (member.getUser().isBot() || member.getUser().isSystem()) return false;
 		if (member.isTimedOut() || member.isPending()) return false;
-		return !hasActiveSubmissionThreads(member.getIdLong());
+		return config.getSubmissionChannel().getThreadChannels()
+				.stream().noneMatch(p -> p.getName().contains(member.getId()));
 	}
 
 	/**
-	 * Checks if the given user has unreviewed submissions.
+	 * Accepts a submission.
 	 *
-	 * @param authorId The user's id.
-	 * @return Whether the user hat unreviewed submissions or not.
+	 * @param hook       The {@link net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent} that was fired.
+	 * @param thread     The submission's {@link ThreadChannel}.
+	 * @param author     The submissions' author.
+	 * @param bestAnswer Whether the submission is among the best answers for this week.
 	 */
-	public boolean hasActiveSubmissionThreads(long authorId) {
-		try (Connection con = dbHelper.getDataSource().getConnection()) {
-			return !qotwSubmissionRepository.getUnreviewedSubmissions(authorId).isEmpty();
-		} catch (SQLException e) {
-			ExceptionLogger.capture(e, getClass().getSimpleName());
-			return false;
+	public void acceptSubmission(InteractionHook hook, @NotNull ThreadChannel thread, @NotNull User author, boolean bestAnswer) {
+		thread.getManager().setName(SUBMISSION_ACCEPTED + thread.getName().substring(1)).queue();
+		pointsService.increment(author.getIdLong());
+		notificationService.withQOTW(thread.getGuild(), author).sendAccountIncrementedNotification();
+		if (bestAnswer) {
+			pointsService.increment(author.getIdLong());
+			notificationService.withQOTW(thread.getGuild(), author).sendBestAnswerNotification();
 		}
+		Responses.success(hook, "Submission Accepted",
+				"Successfully accepted submission by " + author.getAsMention()).queue();
+		notificationService.withQOTW(thread.getGuild()).sendSubmissionActionNotification(author, new QOTWSubmission(thread), bestAnswer ? SubmissionStatus.ACCEPT_BEST : SubmissionStatus.ACCEPT);
+		Optional<ThreadChannel> newestPostOptional = config.getSubmissionsForumChannel().getThreadChannels()
+				.stream().max(Comparator.comparing(ThreadChannel::getTimeCreated));
+		if (newestPostOptional.isPresent()) {
+			ThreadChannel newestPost = newestPostOptional.get();
+			getMessagesByUser(thread, author).thenAccept(messages -> {
+				for (Message message : messages) {
+					if (message.getAuthor().isBot() || message.getType() != MessageType.DEFAULT) continue;
+					WebhookUtil.ensureWebhookExists(newestPost.getParentChannel().asForumChannel(), wh -> {
```

Comment:
I would only call `ensureWebhookExists` once and put the loop in the lambda of it instead of the other way round.

Reply:
fixed